src/rdiff_backup/selection.py

```diff
@@ -235,14 +247,20 @@ def diryield(rpath):
 
             Returns (rpath, num) where num == 0 means rpath should be
             generated normally, num == 1 means the rpath is a directory
-            and should be included iff something inside is included.
+            and should be included if something inside is included.
             """
             for filename in self._listdir_sorted(rpath):
+                # First check if path is excluded by filename to reduce io calls
+                new_rorpath = rpath.new_index_empty(rpath.index + (filename,))
```

Comment:
Few things:
 
* `rpath.new_index_empty` can't return a RORPath, only another RPath, so the name `new_rorpath` isn't a good idea.
* `rpath` is a terrible name for a variable (and I know it's not your due) as it collides with the module name
* `rp` and `new_rp` aren't great but rather common in the code